react-components/src/hooks/use3dRelatedDirectConnections.ts

```diff
@@ -74,3 +79,27 @@ export function use3dRelatedDirectConnections(
     }
   );
 }
+
+function createDeduplicatedListAndIndexMap(
+  viewsWithObjectIndex: Array<readonly [number, Source]>
+): [Source[], Map<number, number>] {
+  const deduplicatedViews: Source[] = [];
+  const deduplicatedViewsIndexMap = new Map<number, number>();
+  viewsWithObjectIndex.forEach(([_index, view], viewIndex) => {
+    const duplicateIndex = deduplicatedViews.findIndex(
```

Comment:
Can we use `Map` for Deduplication check, instead of using `findIndex` on the array for each iteration, you can use a `Map` to store information about the unique views

Reply:
Good suggestion - it also made the code easier to understand :+1: